e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,365 @@
+	nt.T.Log("Start managing the namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Start managing the namespace")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Validate managed service appears on the cluster")
+	if err := nt.Validate("some-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Remove the namespace directory from the repo")
+	nt.RootRepos[configsync.RootSyncName].Remove(fmt.Sprintf("acme/namespaces/%s", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove the namespace from the managed set of namespaces")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Validate managed service disappears from the cluster")
+	if err := nt.ValidateNotFound("some-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Ensure that the unmanaged service remained")
+	if err := nt.Validate("some-other-service", namespace, &corev1.Service{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	someOtherService := &corev1.Service{}
+	if err := nt.Get("some-other-service", namespace, someOtherService); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Delete(someOtherService); err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func unmanageNamespace(nt *nomostest.NT, namespace string) {
+	nt.T.Log("stop managing the system namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/unmanaged-namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Stop managing the namespace")
+	nt.WaitForRepoSyncs()
+}
+
+func TestNamespaceDefaultCanBeManaged(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2)
+	manageNamespace(nt, "default")
+	unmanageNamespace(nt, "default")
+}
+
+func TestNamespaceGatekeeperSystemCanBeManaged(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2)
+
+	/*nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/namespace.gatekeeper-system.yaml", yamlDir), "acme/namespaces/gatekeeper-system/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Create gatekeeper-system namespace")
+	nt.WaitForRepoSyncs()*/
+	nt.MustKubectl("apply", "-f", fmt.Sprintf("%s/reserved_namespaces/namespace.gatekeeper-system.yaml", yamlDir))
+	manageNamespace(nt, "gatekeeper-system")
+
+	/*nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/gatekeeper-system/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Delete gatekeeper-system namespace")
+	nt.WaitForRepoSyncs()*/
+	nt.MustKubectl("delete", "ns", "gatekeeper-system", "--ignore-not-found")
```

Comment:
I suggest moving this to a `Cleanup` block defined between lines 346/347

Reply:
moved to a cleanup block